tests/test.c

```diff
@@ -7138,46 +7138,61 @@ rd_kafka_mock_cluster_t *test_mock_cluster_new(int broker_cnt,
 }
 
 /**
- * @brief Wait that at least \p num matching requests
+ * @brief Get current number of matching requests,
+ *        received by mock cluster \p mcluster, matching
+ *        function \p match , called with opaque \p opaque .
+ */
+static size_t test_mock_get_matching_request_cnt(
+    rd_kafka_mock_cluster_t *mcluster,
+    rd_bool_t (*match)(rd_kafka_mock_request_t *request, void *opaque),
+    void *opaque) {
+        size_t i;
+        size_t request_cnt;
+        rd_kafka_mock_request_t **requests;
+        size_t matching_request_cnt = 0;
+
+        requests = rd_kafka_mock_get_requests(mcluster, &request_cnt);
+
+        for (i = 0; i < request_cnt; i++) {
+                if (match(requests[i], opaque))
+                        matching_request_cnt++;
+        }
+
+        rd_kafka_mock_request_destroy_array(requests, request_cnt);
+        return matching_request_cnt;
+}
+
+/**
+ * @brief Wait that at least \p expected_cnt matching requests
  *        have been received by the mock cluster,
  *        using match function \p match ,
  *        plus \p confidence_interval_ms has passed
  *
- * @param num Number of expected matching request
- * @param confidence_interval_ms Time to wait after \p num matching requests
- *                               have been seen
+ * @param expected_cnt Number of expected matching request
+ * @param confidence_interval_ms Time to wait after \p expected_cnt matching
+ *                               requests have been seen
  * @param match Match function that takes a request and \p opaque
  * @param opaque Opaque value needed by function \p match
  *
  * @return Number of matching requests received.
  */
-int test_mock_wait_maching_requests(
+size_t test_mock_wait_maching_requests(
     rd_kafka_mock_cluster_t *mcluster,
-    int num,
+    size_t expected_cnt,
     int confidence_interval_ms,
     rd_bool_t (*match)(rd_kafka_mock_request_t *request, void *opaque),
     void *opaque) {
-        size_t i;
-        rd_kafka_mock_request_t **requests;
-        size_t request_cnt;
-        int matching_requests = 0;
-        rd_bool_t last_time   = rd_true;
+        size_t matching_request_cnt = 0;
 
-        while (matching_requests < num || last_time) {
-                if (matching_requests >= num) {
-                        rd_usleep(confidence_interval_ms * 1000, 0);
-                        last_time = rd_false;
-                }
-                requests = rd_kafka_mock_get_requests(mcluster, &request_cnt);
-                matching_requests = 0;
-                for (i = 0; i < request_cnt; i++) {
-                        if (match(requests[i], opaque))
-                                matching_requests++;
-                }
-                rd_kafka_mock_request_destroy_array(requests, request_cnt);
-                rd_usleep(100 * 1000, 0);
+        while (matching_request_cnt < expected_cnt) {
```

Comment:
If this criteria is never met, the test will be in infinite loop. Can we make sure that there is some deterministic terminate condition?